core/sds-aspect-model-aas-generator/src/test/java/io/openmanufacturing/sds/aspectmodel/aas/AspectModelAASGeneratorTest.java

```diff
@@ -0,0 +1,276 @@
+            "urn:bamm:io.openmanufacturing.test:1.0.0#TestAspectintRcProperty",
+            "urn:bamm:io.openmanufacturing.test:1.0.0#TestAspectbigIntRcProperty",
+            "urn:bamm:io.openmanufacturing.test:1.0.0#TestAspectfloatRcProperty",
+            "urn:bamm:io.openmanufacturing.test:1.0.0#TestAspectstringRegexcProperty")
+            .collect(Collectors.toCollection(HashSet::new));
+
+      checkDataSpecificationIEC61360(semanticIds, env);
+   }
+
+   @Test
+   void testGenerateAasxFromBammAspectWithRecursivePropertyWithOptional() throws IOException, DeserializationException {
+      AssetAdministrationShellEnvironment env = getAssetAdministrationShellFromAspect( TestAspect.ASPECT_WITH_RECURSIVE_PROPERTY_WITH_OPTIONAL );
+      assertEquals( 1, env.getSubmodels().size(), "Not exactly one Submodel in AAS." );
+   }
+
+   @Test
+   void testGenerateAasxFromBammAspectWithCode() throws IOException, DeserializationException {
+      AssetAdministrationShellEnvironment env = getAssetAdministrationShellFromAspect( TestAspect.ASPECT_WITH_CODE );
+      assertEquals( 2, env.getConceptDescriptions().size() );
+      assertEquals( 1, env.getSubmodels().size() );
+      assertEquals( 1, env.getSubmodels().get( 0 ).getSubmodelElements().size() );
+      Property submodelElement = (Property) env.getSubmodels().get( 0 ).getSubmodelElements().get( 0 );
+      assertEquals( "http://www.w3.org/2001/XMLSchema#int", submodelElement.getValueType(), "Value type not int" );
+
+      DataSpecificationIEC61360 dataSpecificationContent =
+            getDataSpecificationIEC61360( "TestCode", env );
+   }
+
+   @Test
+   void testGenerateAasxFromBammAspectWithEnumeration() throws IOException, DeserializationException {
+      AssetAdministrationShellEnvironment env = getAssetAdministrationShellFromAspect( TestAspect.ASPECT_WITH_ENUMERATION );
+
+      assertEquals( 2, env.getConceptDescriptions().size() );
+
+      DataSpecificationIEC61360 dataSpecificationContent = (DataSpecificationIEC61360)
+            env.getConceptDescriptions().stream().filter( x -> x.getIdShort().equals( "TestEnumeration" ) ).findFirst().get()
+                  .getEmbeddedDataSpecifications().stream().findFirst().get( )
+                  .getDataSpecificationContent();
+      assertEquals( 3, dataSpecificationContent.getValueList().getValueReferencePairTypes().size());
+
+      assertEquals( 1, env.getSubmodels().size() );
+      assertEquals( 1, env.getSubmodels().get( 0 ).getSubmodelElements().size() );
+      Property submodelElement = (Property) env.getSubmodels().get( 0 ).getSubmodelElements().get( 0 );
+      assertEquals( "http://www.w3.org/2001/XMLSchema#integer", submodelElement.getValueType(), "Value type not int" );
+   }
+
+   @ParameterizedTest
+   @EnumSource( value = TestAspect.class, mode = EnumSource.Mode.EXCLUDE, names = {
+         "ASPECT_WITH_ABSTRACT_ENTITY", // Aspect model not present
```

Comment:
These Aspect models are not found because the `TestAspect` enum refers to the set of test models that are present for the latest meta model version (the latest and greatest set). This fails here because the `loadAspect()` method has loading test models for BAMM 1.0.0 hard coded:
`TestResources.getModel( testAspect, KnownVersion.BAMM_1_0_0 ).get()`
This should be replace by the following:
`TestResources.getModel( testAspect, KnownVersion.getLatest() ).get()`

Then the excludes here can be removed!